src/storm/mesos/MesosNimbus.java

```diff
@@ -617,6 +608,56 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
       }
     }
   }
+  
+  private ExecutorInfo createExecutorInfo(Optional<ContainerInfo.Builder> containerInfoBuilder,
+          TopologyDetails details, String executorDataStr, double executorCpu, double executorMem, 
+          String cpuRole, String memRole, String configUri) {
+              
+        ExecutorInfo.Builder infoBuilder = ExecutorInfo.newBuilder()
+                .setExecutorId(ExecutorID.newBuilder().setValue(details.getId()))
+                .setData(ByteString.copyFromUtf8(executorDataStr))
+                .addResources(Resource.newBuilder()
```

Comment:
I will leave that in your capable hands since you seem to understand the problem and know what to change. Thanks.